app/models/concerns/orchestration/tftp.rb

```diff
@@ -42,16 +42,19 @@ def generate_pxe_template
     # work around for ensuring that people can use @host as well, as tftp templates were usually confusing.
     @host = self.host
     if build?
-      pxe_render host.provisioning_template({:kind => host.operatingsystem.template_kind})
+      template = host.provisioning_template({:kind => host.operatingsystem.template_kind})
+      template_name = template.name
     else
       if host.operatingsystem.template_kind == "PXEGrub"
-        pxe_render ProvisioningTemplate.find_by_name("PXEGrub default local boot")
+        template_name = "PXEGrub default local boot"
       else
-        pxe_render ProvisioningTemplate.find_by_name("PXELinux default local boot")
+        template_name = "PXELinux default local boot"
       end
+      template = ProvisioningTemplate.find_by_name(template_name)
     end
+    unattended_render template, template_name
   rescue => e
-    failure _("Failed to generate %{template_kind} template: %{e}") % { :template_kind => host.operatingsystem.template_kind, :e => e }, e
+    failure _("Failed to generate '%{name}' template: %{e}") % { :name => template_name, :e => e }, e
```

Comment:
I think it'll worsen the error message when a specific template kind is missing - it would previously have said the template kind (Failed to generate PXELinux template). It shouldn't be changed to 'nil' in that case.  I'd suggest moving the rescue block to surround only the `unattended_render`.